### ISSUE TITLE
`code-block-tools`: add `on-unknown-language-tag` for unrecognized fenced code block languages

### ISSUE THREAD

**Issue by pygarap** [rvben/rumdl#894]
## Summary

Add an `on-unknown-language-tag` option to Code Block Tools to control what happens when a fenced code block has a non-empty language tag that rumdl cannot recognize or resolve.

This is distinct from `on-missing-language-definition`: an unknown tag such as `pyhton` is invalid, while a recognized language such as `python` may simply have no tools configured.

`normalize-language = "exact"` does not solve this because it only disables language alias normalization. It does not validate whether a fence tag is recognized.

## Proposed behavior

```toml
[code-block-tools]
on-unknown-language-tag = "warn"
```

| Value | Behavior |
| --- | --- |
| `"warn"` | Skip the affected block, report each distinct unknown language tag once per run, and continue (default) |
| `"ignore"` | Silently skip affected blocks |
| `"fail"` | Record an error for each affected block, continue processing, and exit non-zero at the end |
| `"fail-fast"` | Stop immediately on the first affected block and exit non-zero |

A tag should count as recognized when it matches GitHub Linguist, a configured `language-aliases` entry, or an explicitly configured custom language.

The check should happen before `on-missing-language-definition`, so recognized but unconfigured languages continue to use that existing option.

This is particularly useful with `--only-code-block-tools`, where regular Markdown rules such as MD040 are intentionally excluded.

## Usage

```toml
[code-block-tools]
enabled = true
on-unknown-language-tag = "fail"
on-missing-language-definition = "fail"

[code-block-tools.languages]
python = { lint = ["ruff:check"], format = ["ruff:format"] }
```

Given:

```markdown
# Example

~~~pyhton
print("Hello")
~~~
```

Run:

```console
$ rumdl check --only-code-block-tools docs/
```

With `on-unknown-language-tag = "fail"`, rumdl should report `pyhton` as an unrecognized language tag instead of treating it as merely an unconfigured language.

## References

- https://github.com/rvben/rumdl/blob/main/docs/code-block-tools.md
- https://github.com/rvben/rumdl/blob/main/docs/md040.md
- #374
- #377
- #378
- #889
